R/estimate_incubation.R

```diff
@@ -0,0 +1,67 @@
+#' Function to extract the empirical incubation period distribution from line list data.
+#' Can take into account uncertain dates of exposure.
+#'
+#' @param x the linelist data (data.frame or linelist object) containing at least a column containing the exposure dates and one containing the onset dates. For exposure dates, each element can be a list containing several possible exposure dates.
+#' @param dates_exposure the name of the column containing the exposure dates.
+#' @param date_onset the name of the column containing the onset dates
+#' @return a data frame containing a column with the different incubation periods and a column containing their relative frequency
+#' @author Flavio Finger, \email{flavio.finger@lshtm.ac.uk}
+#' @export
+empirical_incubation_dist  <- function(x, dates_exposure, date_of_onset) {
+  #error checking
+  if (!is.data.frame(x)) {
+    stop("x is not a data.frame")
+  }
+
+  if (ncol(x)==0L) {
+    stop("x has no columns")
+  }
+
+  dates_exposure <- rlang::enquo(dates_exposure)
+  date_of_onset <- rlang::enquo(date_of_onset)
+
+  y <- x %>%
```

Comment:
I think this whole block needs to be moved into a separate internal function that handles two date vectors. That way, it will be easier to debug later on.